Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1

```diff
@@ -64,12 +65,25 @@ if ($enableDetailedLoggingString -ne "true")
 function ThrowError
 {
     param([string]$errorMessage)
-    
+
         $readmelink = "http://aka.ms/powershellontargetmachinesreadme"
         $helpMessage = (Get-LocalizedString -Key "For more info please refer to {0}" -ArgumentList $readmelink)
         throw "$errorMessage $helpMessage"
 }
 
+function Write-Telemetry
+{
+  [CmdletBinding()]
+  param(
+    [Parameter(Mandatory=$True,Position=1)]
+    [string]$code
+    )
+
+  $telemetryString = "##vso[task.logissue type=error;code=" + $code + ";TaskId=3B5693D4-5777-4FEE-862A-BD2B7A374C68;]"
+  Write-Host $telemetryString
+  $global:telemetrySet = $true
```

Comment:
It is probably ok to log telemetry twice for a scenario where a prereq fails since it is logical that the predep error code can contain one or more prereqs. Dependency on a global scope variable will create unintentional side effects in other tasks?